### PR TITLE
Environment variables and secrets refactor

### LINKED ISSUE #986: Better Secrets.swift setup
We've tried this before but haven't landed on something simple. Here's what I'd love to see:

- Set environment variables on command line
- Compiler **fails** when variables haven't been set

Then it should be basically impossible to screw up shipping a beta or App Store build w/out keys.

### PR DESCRIPTION
**Pull request by rnystrom**
Using #197 as a jumping point, I added a new secrets setup that has:

- Ignored, generated secrets config. Update once, never touch again
- Config files are added to Xcode project file, so CI should build
- Add secrets to buddybuild as env variables

Fixes #986

### CODE REVIEW

**Comment by rnystrom** [comment]
CI is green!

https://dashboard.buddybuild.com/apps/59d904ff8588d60001729c4e/build/5a11fb10f1c38000011ac58d

File: Classes/Systems/Secrets.m [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+//  Secrets.m
+//  Freetime
+//
+//  Created by Ryan Nystrom on 11/19/17.
+//  Copyright © 2017 Ryan Nystrom. All rights reserved.
+//
+
+#import "Secrets.h"
+
+#define STRINGIFY(x) @#x
```
Comment (Sherlouk) [inline Classes/Systems/Secrets.m]: Not exactly sure what this is, but assuming this is environment variables as per your comment - could we use (in Swift) `ProcessInfo.processInfo.environment[x]` instead?

File: .gitignore [hunk 2]
```diff
@@ -74,3 +74,7 @@ node_modules
 _site/
 .sass-cache/
 .jekyll-metadata
+
+# secrets
+Resources/*.xcconfig
```
Comment (Sherlouk) [inline .gitignore]: Should we commit an example xcconfig to know the format for those unfamiliar?

Comment (rnystrom) [reply .gitignore]: No worries! The script generates the files for you

File: Classes/Systems/Secrets.m (hunk 1 above)
Comment (rnystrom) [reply Classes/Systems/Secrets.m]: Hmm maybe, just following #197 atm

File: .gitignore (hunk 2 above)
Comment (Sherlouk) [reply .gitignore]: Ahh cool, sorry haven't run it yet!

File: Classes/Systems/Secrets.m (hunk 1 above)
Comment (Sherlouk) [reply Classes/Systems/Secrets.m]: Mmm, alright I'm not too precious about it - just my lack of ObjC knowledge 😅

Comment (rnystrom) [reply Classes/Systems/Secrets.m]: Lol ya, same, its gonna be hacks no matter how we slice it. I'm def in uncharted territory atm lol.

File: .gitignore (hunk 2 above)
Comment (BasThomas) [reply .gitignore]: Should we be ignoring only `xcconfig`s in the Resources folder?

**Comment by BasThomas** [comment]
Can you please add some documentation on how to set this up the first time? I've edited the `xcconfig`, but for whatever reason it still inserts `GITHUB_CLIENT_ID` in the callback URL.

**Comment by rnystrom** [comment]
Yes! Will document better

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Pull request merged**